basic-list-techniques/mapping.prolog

```diff
@@ -0,0 +1,43 @@
+:- module(basic_list_techniques_mapping, []).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+square(X, R) :-
+    R is X * X.
+
+cube(X, R) :-
+    R is X * X * X.
+
+map_element(X, XMapped) :-
+    square(X, XSquare),
+    cube(X, XCube),
+    XMapped = [X, square, XSquare, cube, XCube].
+
+map([], []).
+
+map([Head|Tail], Mapped) :-
+    map(Tail, TailMapped),
+    map_element(Head, HeadMapped),
+    Mapped = [HeadMapped|TailMapped].
```

Comment:
This line is redundant and should be replaced with `[Head Mapped| TailMapped]` in the head.